Content type updates for Framework migration guide (#602)
hashicorp/web-unified-docs: content/terraform-plugin-framework/v1.15.x/docs/plugin/framework/migrating/attributes-blocks/default-values.mdx

---
page_title: Migrating attribute default values
description: >-
  Learn how to migrate attribute default values from SDKv2 by using an attribute
  plan modifier in the Terraform plugin framework.
---

# Migrating attribute default values

This page explains how to migrate attribute defaults in SDKv2 to the framework.

## Background

Default values set a value for an attribute when the Terraform configuration

> does not provide one. In SDKv2 and the framework default values are set via the
> `Default` field on an attribute's schema. Refer to
> [Default](/terraform/plugin/framework/resources/default) in the framework
> documentation for details.

Reviewer: ```suggestion
does not provide one. Use the `Default` field on an attribute's schema to 
set default values in SDKv2 and the framework. Refer to 
[Default](/terraform/plugin/framework/resources/default) in the framework
documentation for details.
```

Default values support is only available in the framework for resource
attributes. Handle default values for data source attributes within the [data
source `Read` method](/terraform/plugin/framework/data-sources#read-method) and
default values for provider attributes within the [provider `Configure`
method](/terraform/plugin/framework/providers#configure-method).

## Migrate an attribute with a default value

> In SDKv2, default values are defined for a primitive attribute type (i.e.,
> `TypeBool`, `TypeFloat`, `TypeInt`, `TypeString`) by the `Default` field on the
> attribute's schema. Alternatively, the `DefaultFunc` function is used to compute
> a default value for an attribute.

Reviewer: ```suggestion
In SDKv2, you define default values for a primitive attribute type by the `Default` field on the attribute's schema.
These primitive attriutes include the `TypeBool`, `TypeFloat`, `TypeInt`, `TypeString` types. 
 Alternatively, the you can use the `DefaultFunc` function to compute
a default value for an attribute.
```
